client/material/tooltip.tsx

```diff
@@ -0,0 +1,255 @@
+   * */
+  tooltipContainer?: React.ReactNode
+}
+
+/**
+ * A component that displays some content in a floated UI element when a user hovers over a target
+ * element. Should generally be used over native `title` attribute on elements.
+ *
+ * Utilizes popovers to deal with positioning, but with a much simpler API than that of a Popover.
+ */
+export function Tooltip({
+  text,
+  children,
+  position = 'bottom',
+  className,
+  tooltipContainer,
+}: TooltipProps) {
+  const [open, setOpen] = useState(false)
+  const [anchorElem, setAnchorElem] = useState<HTMLElement>()
+  const transition = useTransition<boolean, UseTransitionProps<boolean>>(open, {
+    from: { opacity: 0, scale: 0.667 },
+    enter: { opacity: 1, scale: 1 },
+    leave: { opacity: 0, scale: 0.333 },
+    config: (item, index, phase) => key =>
+      phase === 'leave' || key === 'opacity' ? { ...defaultSpring, clamp: true } : defaultSpring,
+  })
+
+  const anchorOriginX = position === 'top' || position === 'bottom' ? 'center' : position
+  const anchorOriginY = position === 'left' || position === 'right' ? 'center' : position
+  let [, anchorX = 0, anchorY = 0] = useAnchorPosition(
+    anchorOriginX,
+    anchorOriginY,
+    anchorElem ?? null,
+  )
+
+  const onMouseEnter = useCallback((event: React.MouseEvent) => {
+    setAnchorElem(event.currentTarget as HTMLElement)
+  }, [])
+  const onMouseLeave = useCallback((event: React.MouseEvent) => {
+    setAnchorElem(undefined)
+  }, [])
+
+  useEffect(() => {
+    if (anchorElem) {
+      let timeout: ReturnType<typeof setTimeout> | undefined = setTimeout(() => {
+        timeout = undefined
+        setOpen(true)
+      }, 200)
+
+      return () => {
+        if (timeout) {
+          clearTimeout(timeout)
+        }
+      }
+    } else {
+      setOpen(false)
+      return () => {}
+    }
+  }, [anchorElem])
+
+  let originX: OriginX
+  if (anchorOriginX === 'left') {
+    originX = 'right'
+    anchorX = anchorX - 8
+  } else if (anchorOriginX === 'right') {
+    originX = 'left'
+    anchorX = anchorX + 8
+  } else {
+    originX = 'center'
+  }
+
+  let originY: OriginY
+  if (anchorOriginY === 'top') {
+    originY = 'bottom'
+    anchorY = anchorY - 8
+  } else if (anchorOriginY === 'bottom') {
+    originY = 'top'
+    anchorY = anchorY + 8
+  } else {
+    originY = 'center'
+  }
+
+  return (
+    <>
+      <TooltipChildrenWrapper
+        text={text}
+        className={className}
+        onMouseEnter={onMouseEnter}
+        onMouseLeave={onMouseLeave}>
+        {children}
+      </TooltipChildrenWrapper>
+      {transition(
+        (styles, open) =>
+          open && (
+            <NoPointerPortal open={open}>
+              <NoPointerPopoverContent
+                anchorX={anchorX}
+                anchorY={anchorY}
+                originX={originX}
+                originY={originY}
+                styles={styles}>
+                {tooltipContainer ? (
+                  tooltipContainer
```

Comment:
I don't think it makes sense for this to work this way, it should be treated as a replacement for TooltipContent and receive the same props (`$position`, children). If the replacer doesn't care about those things they can ignore them, if they do it prevents them having to re-implement that stuff themselves.